packages/components/src/external-link/index.tsx

```diff
@@ -32,12 +32,22 @@ function UnforwardedExternalLink(
 		compact( [ ...rel.split( ' ' ), 'external', 'noreferrer', 'noopener' ] )
 	).join( ' ' );
 	const classes = classnames( 'components-external-link', className );
+	/* Anchor links are percieved as external links.
+	This constant helps check for on page anchor links,
+	to prevent them from being opened in the editor. */
+	const shouldPreventDefault = href.startsWith( '#' );
```

Comment:
```suggestion
	const isInternalAnchor = href.startsWith( '#' );
```

Perhaps we should rename the var to describe what it represents?